### PR TITLE
[Datasource] Add Hostgroups, Hostgroup

### PR DESCRIPTION
**Pull request by michaelkad**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Output from acceptance testing:

```
=== RUN   TestAccIBMPiHostgroupsDataSourceBasic
--- PASS: TestAccIBMPiHostgroupsDataSourceBasic (12.85s)
PASS

=== RUN   TestAccIBMPIHostgroupDataSourceBasic
--- PASS: TestAccIBMPIHostgroupDataSourceBasic (10.32s)
PASS
```

### CODE REVIEW

File: website/docs/d/pi_hostgroups.html.markdown [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  Get information about hostgroups.
+---
+
+# ibm_pi_hostgroups
+
+Provides a read-only data source to retrieve information about hostgroups. you can use in Power Systems Virtual Server. For more information, about ower Systems Virtual Server hostgroup, see [hostgroups](https://cloud.ibm.com/apidocs/power-cloud#endpoint).
+
+## Example Usage
+
+```hcl
```
Comment (ismirlia) [inline website/docs/d/pi_hostgroups.html.markdown:15]: hcl?

File: website/docs/d/pi_hostgroups.html.markdown [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+In addition to all argument reference list, you can access the following attribute references after your data source is created.
+
+- `hostgroups` - (List) List of hostgroups.
+  - Nested sheme for `hostgroups`:
+        - `creation_date` - (String) Date/Time of hostgroup creation.
+        - `id` - (String) The unique identifier of the hostgroup.
+        - `hosts` - (List) List of hosts.
+        - `name` - (String) Name of the hostgroup.
+        - `primary` - (String) Name of the workspace owning the hostgroup.
+        - `secondaries` - (List) Names of workspaces the hostgroup has been shared with
```
Comment (ismirlia) [inline website/docs/d/pi_hostgroups.html.markdown:54]: missing period?

File: ibm/provider/provider.go [hunk 3]
```diff
@@ -649,6 +649,7 @@ func Provider() *schema.Provider {
 			"ibm_pi_workspaces":                             power.DatasourceIBMPIWorkspaces(),
 			"ibm_pi_datacenter":                             power.DataSourceIBMPIDatacenter(),
 			"ibm_pi_datacenters":                            power.DataSourceIBMPIDatacenters(),
+			"ibm_pi_hostgroups":                             power.DataSourceIBMPiHostgroups(),
```
Comment (ismirlia) [inline ibm/provider/provider.go:652]: Can you sort these with the rest of the data sources?

File: website/docs/d/pi_host_groups.html.markdown [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+In addition to all argument reference list, you can access the following attribute references after your data source is created.
+
+- `host_groups` - (List) List of hostgroups.
+  - Nested sheme for `host_groups`:
+    - `creation_date` - (String) Date/Time of host group creation.
+    - `id` - (String) The unique identifier of the host group.
+    - `hosts` - (List) List of hosts.
+    - `name` - (String) Name of the host group.
+    - `primary` - (String) Name of the workspace owning the host group.
+    - `secondaries` - (List) Names of workspaces the host group has been shared with.
```
Comment (ismirlia) [inline website/docs/d/pi_host_groups.html.markdown:54]: I think this description has been updated say workspace IDs in the model.

File: ibm/service/power/data_source_ibm_pi_host_groups.go [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+						},
+						Attr_Hosts: {
+							Computed:    true,
+							Description: "List of hosts.",
+							Elem: &schema.Schema{
+								Type: schema.TypeString,
+							},
+							Type: schema.TypeList,
+						},
+						Attr_ID: {
```
Comment (ismirlia) [inline ibm/service/power/data_source_ibm_pi_host_groups.go:50]: Is this value not computed as well?

**Review by ismirlia — APPROVED** [review]

File: ibm/service/power/data_source_ibm_pi_host_group_test.go [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+			},
+		},
+	})
+}
+
+func testAccCheckIBMPIHostgroupDataSourceConfigBasic() string {
+	return fmt.Sprintf(`
+		data "ibm_pi_host_group" "hostGroup" {
+			pi_cloud_instance_id = "%s"
+			pi_hostgroup_id  = "%s"
```
Comment (michaelkad) [inline ibm/service/power/data_source_ibm_pi_host_group_test.go:34]:
```suggestion
			pi_host_group_id  = "%s"
```

File: website/docs/d/pi_host_group.html.markdown [hunk 7]
```diff
@@ -0,0 +1,59 @@
+---
+subcategory: "Power Systems"
+layout: "ibm"
+page_title: "IBM: ibm_pi_host_group"
+description: |-
+  Manages a host group in
```
Comment (michaelkad) [inline website/docs/d/pi_host_group.html.markdown:6]:
```suggestion
  Get information about a host group
```

File: website/docs/d/pi_host_groups.html.markdown [hunk 8]
```diff
@@ -0,0 +1,54 @@
+---
+subcategory: "Power Systems"
+layout: "ibm"
+page_title: "IBM: ibm_pi_host_groups"
+description: |-
+  Get information about hostgroups.
```
Comment (michaelkad) [inline website/docs/d/pi_host_groups.html.markdown:6]:
```suggestion
  Get information about host groups.
```

File: website/docs/d/pi_host_groups.html.markdown [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+
+Review the argument references that you can specify for your data source.
+
+- `pi_cloud_instance_id` - (Required, String) The GUID of the service instance associated with an account.
+
+## Attribute Reference
+
+In addition to all argument reference list, you can access the following attribute references after your data source is created.
+
+- `host_groups` - (List) List of hostgroups.
```
Comment (michaelkad) [inline website/docs/d/pi_host_groups.html.markdown:47]:
```suggestion
- `host_groups` - (List) List of host groups.
```

**Comment by michaelkad** [comment]
Merged

**Pull request closed without merging**